### ISSUE TITLE
Color sorting inversed for first and last item in ColorRange.getColor()

### ISSUE THREAD

**Issue by ross-campbell** [Heyeso/color-range#22]
Hey, nice little library i am finding useful.

I have been running into one issue though when using the tool.

So, say i want 14 different shades of blue ordered from light to dark. 
I have used code similar to the below to create a map of 14 colors.

However, when querying the colorRange(test_colors, test_ranges).getColor(i).toHex method, all items are returned in the expected light to dark order, aside from the first and last item.

```
let startRange = 1;
let endRange = 14;

const test_colors = [
      [204, 255, 255],
      [0, 51, 51],
    ];

const test_ranges = [startRange, endRange];

for (let i = startRange; i <= endRange; i++) {
    console.log(colorRange(test_colors, test_ranges).getColor(i).toHex);
}

```

This image illustrates the order in which the colors are returned.
![image](https://github.com/Heyeso/color-range/assets/45824634/c74c41da-11fc-46ef-915e-adc049a8e757)

After a bit more testing, i found that in the [color-range.ts](https://github.com/Heyeso/color-range/blob/main/src/color-range.ts) file (line 89), if i reverse the two return statements on the two if's like below:
```
// if (num <= this.map.ranges[0]) return this.map.colors[0];
// if (num >= this.map.ranges[this.map.ranges.length - 1])
    //   return this.map.colors[this.map.ranges.length - 1];

if (num <= this.map.ranges[0]) return this.map.colors[this.map.ranges.length - 1];
if (num >= this.map.ranges[this.map.ranges.length - 1]) return this.map.colors[0];
```
Then all 14 colors are returned in the expected order including the first and last item. 
This also occurs when the order of the base colors in "test_colors" are also revered from Light to Dark, to, Dark to Light

**Comment by Heyeso** [comment]
Hello! Great catch there. Feel Free to submit a pull request.

**Comment by Heyeso** [comment]
The issue is from [line 15 in `/src/utils.ts`](https://github.com/Heyeso/color-range/blob/main/src/utils.ts#L15C1-L15C61). `ratio` refers to the amount of `color1` we **do not** want, and the amount of `color2` we want.

Replacing the `blendChannelByRatio` function with the code below should fix it.
```ts
export const blendChannelByRatio = (
  channel1: number,
  channel2: number,
  ratio: number
): number =>  (channel1 * (1 - ratio)) + (channel2 * ratio);
```

**Comment by Heyeso** [comment]
Fixed in [@heyeso/color-range@3.0.1](https://www.npmjs.com/package/@heyeso/color-range)

**Comment by ross-campbell** [comment]
Hey, Thanks for the quick fix!
I can confirm that all colors are now returned in the expected order in v3.0.1
